aws/iam.go

```diff
@@ -0,0 +1,66 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// List all IAM users in the AWS account and returns a slice of the UserNames
+// TODO: Implement exclusion by time filter
+func getAllIamUsers(session *session.Session, configObj config.Config) ([]*string, error) {
+	svc := iam.New(session)
+	input := &iam.ListUsersInput{}
+
+	var userNames []*string
+
+	// TODO: Probably use ListUsers together with ListUsersPages in case there are lots of users
```

Comment:
Question: is there any number you're thinking of here? You could also probably use `Marker` and `MaxItems` - see [here](https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListUsers.html).

Reply:
No, I wasn't thinking about any numbers specifically. Just because of the "mantra/practice/scars" of _"if something have pagination, it will use pagination"_ :stuck_out_tongue:

Reply:
Hahah, that makes sense :)